src/matter/cluster/ScenesCluster.ts

```diff
@@ -0,0 +1,293 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    ArrayT,
+    BooleanT,
+    Field,
+    OptionalField,
+    ObjectT,
+    StringT,
+    UInt8T,
+    UInt16T,
+    UInt32T,
+    Bound,
+    EnumT, BitMapT, Bit
+} from "../../codec/TlvObjectCodec";
+import { Attribute, OptionalAttribute, Cluster, Command, OptionalCommand, NoResponseT } from "./Cluster";
+import { StatusCode } from "../interaction/InteractionMessages";
+import { GroupIdT } from "../common/GroupId";
+import { ClusterIdT } from "../common/ClusterId";
+import { NodeIdT } from "../common/NodeId";
+import { AttributeIdT } from "../common/AttributeId";
+import { MatterApplicationClusterSpecificationV1_0 } from "../../Specifications";
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.4.6.1 */
+const AttributeValuePairT = ObjectT({
+    /**
+     * This field SHALL be present or not present, for all instances in the Scenes cluster. If this field is
+     * not present, then the data type of AttributeValue SHALL be determined by the order and data type defined
+     * in the cluster specification. Otherwise the data type of AttributeValue SHALL be the data type of the
+     * attribute indicated by AttributeID.
+     */
+    attributeId: OptionalField(0, AttributeIdT),
+
+    /** This is the attribute value as part of an extension field set. */
+    attributeValue: Field(1, ArrayT(UInt32T)), /* TODO: type wise would be "ArrayT(unknown)", but ... :-) */
```

Comment:
If the type is not known but will be known at runtime, you should use AnyT

This expect the caller to already have transform this field using the right template before.
See DataReportT in InteractionMessage.ts as an example

Reply:
adjusted